### PR TITLE
🌱 Initial forum mocks and pages

### PR DESCRIPTION
**Pull request by jodator**
See #1061. This PR introduces:

*   basic forum categories & forum threads types,
*   mocks generators & db seeds
*   basic queries
*   basic hooks
*   base pages for forum

### CODE REVIEW

File: packages/ui/src/forum/types/ForumCategory.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  description: string
+}
+
+export const asForumCategory = (fields: ForumCategoryFieldsFragment): ForumCategory => {
+  return {
+    id: fields.id,
+    title: fields.title,
+    description: fields.description,
+  }
+}
```
Comment (DevNikita16) [inline packages/ui/src/forum/types/ForumCategory.ts:15]:
```suggestion
export const asForumCategory = (fields: ForumCategoryFieldsFragment): ForumCategory => ({
    id: fields.id,
    title: fields.title,
    description: fields.description,
})
```

Comment (DevNikita16) [reply packages/ui/src/forum/types/ForumCategory.ts:15]: Not a problem, but I think it looks better 👀

File: packages/ui/src/forum/types/ForumThread.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export const asForumThread = (fields: ForumThreadFieldsFragment): ForumThread => {
+  return {
+    id: fields.id,
+    title: fields.title,
+    isSticky: fields.isSticky,
+    categoryId: fields.categoryId,
+  }
+}
```
Comment (DevNikita16) [inline packages/ui/src/forum/types/ForumThread.ts:17]: Same here

**Review by DevNikita16 — APPROVED** [review]

**Comment by DevNikita16** [comment]
Overall looks fine 🚀

File: packages/ui/src/forum/types/ForumCategory.ts (hunk 1 above)
Comment (jodator) [reply packages/ui/src/forum/types/ForumCategory.ts:15]: Yep, I'll update that.

**Review by p-sad — APPROVED** [review]

**Pull request merged**